node-tests/index-test.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-sync */
+'use strict';
+
+const fs = require('fs-extra');
+const path = require('path');
+const exec = require('child_process').exec;
+const assert = require('yeoman-assert');
+
+const TEST_TIMEOUT = 50000;
+const MOCK_ENV_CONFIGS = {
+	default: path.resolve(__dirname, './mocks/environment-default.js'),
+	modules: path.resolve(__dirname, './mocks/environment-modules.js'),
+	noAutoImport: path.resolve(__dirname, './mocks/environment-no-autoimport.js')
+};
+
+const emberCLIPath = path.resolve(__dirname, '../node_modules/ember-cli/bin/ember');
+const fixturePath = path.resolve(__dirname, '../');
+const fixtureConfigPath = path.resolve(__dirname, '../tests/dummy/config');
+const distPath = path.resolve(__dirname, '../dist');
+
+function runEmberCommand(packagePath, command) {
+	return new Promise((resolve, reject) =>
+		exec(`${emberCLIPath} ${command}`, {
+			cwd: packagePath
+		}, (err, result) => {
+			if (err) {
+				reject(err);
+			}
+			resolve(result);
+		})
+	);
+}
+
+function mockConfig(mockFile) {
```

Comment:
Try not to use `sync` functions and use `async/await` instead , you disabled it at the first line of the line.

Reply:
I think using `async/await` this way has the same result that using the sync versions of `fs` and it's a rule inherited from `eslint-plugin-node` that is ok not to follow for this kind of operations. I'm not making server requests and I really want file system operations to be synchronous. Also, using `async/await` requires at least Node 7.6 and the addon specifies that is supports version 6.* 

I'm changing it to not use sync methods, but I think that there is no improvement with this change and test files are allowed to not follow the [`no-sync` rule](https://eslint.org/docs/rules/no-sync).

Reply:
Changed in b3f6c4c

Reply:
Node 6 has been deprecated, now we must use node 8 which has `async/await` native.